pyts/classification/bosssp.py

```diff
@@ -0,0 +1,270 @@
+            X_bow = np.asarray([' '.join(X_word[i, not_equal[i]])
+                                for i in range(n_samples)])
+        else:
+            X_bow = np.asarray([' '.join(X_word[i]) for i in range(n_samples)])
+
+        words_list = np.asarray(X_bow[0].split(' ')).tolist()
+        return words_list
+
+    def decision_function(self, X):
+        pass
+
+    def predict(self, X):
+        pass
+
+    def _check_params(self, n_timestamps):
+        if not isinstance(self.word_size, (int, np.integer)):
+            raise TypeError("'word_size' must be an integer.")
+        if not self.word_size >= 1:
+            raise ValueError("'word_size' must be a positive integer.")
+
+        if not isinstance(self.window_size,
+                          (int, np.integer, float, np.floating)):
+            raise TypeError("'window_size' must be an integer or a float.")
+        if isinstance(self.window_size, (int, np.integer)):
+            if self.drop_sum:
+                if not 1 <= self.window_size <= (n_timestamps - 1):
+                    raise ValueError(
+                        "If 'window_size' is an integer, it must be greater "
+                        "than or equal to 1 and lower than or equal to "
+                        "(n_timestamps - 1) if 'drop_sum=True'."
+                    )
+            else:
+                if not 1 <= self.window_size <= n_timestamps:
+                    raise ValueError(
+                        "If 'window_size' is an integer, it must be greater "
+                        "than or equal to 1 and lower than or equal to "
+                        "n_timestamps if 'drop_sum=False'."
+                    )
+            window_size = self.window_size
+        else:
+            if not 0 < self.window_size <= 1:
+                raise ValueError(
+                    "If 'window_size' is a float, it must be greater "
+                    "than 0 and lower than or equal to 1."
+                )
+            window_size = ceil(self.window_size * n_timestamps)
+
+        if not isinstance(self.window_step,
+                          (int, np.integer, float, np.floating)):
+            raise TypeError("'window_step' must be an integer or a float.")
+        if isinstance(self.window_step, (int, np.integer)):
+            if not 1 <= self.window_step <= n_timestamps:
+                raise ValueError(
+                    "If 'window_step' is an integer, it must be greater "
+                    "than or equal to 1 and lower than or equal to "
+                    "n_timestamps."
+                )
+            window_step = self.window_step
+        else:
+            if not 0 < self.window_step <= 1:
+                raise ValueError(
+                    "If 'window_step' is a float, it must be greater "
+                    "than 0 and lower than or equal to 1."
+                )
+            window_step = ceil(self.window_step * n_timestamps)
```

Comment:
Since this method is run in the `_boss_word_extractor` method, if the values of `window_size` and `window_step` are floats, the proportions are computed on the split datasets, which means that the values of `window_size` and `window_step` would be different for different levels. It might not be an issue in itself, but in this case it should probably be mentioned in the documentation.